-------------------------------------------------------------------------------
Test set: poly.edu.vn.display.other.ChuyendeDaoNGTest
-------------------------------------------------------------------------------
Tests run: 9, Failures: 7, Errors: 0, Skipped: 0, Time elapsed: 4.454 s <<< FAILURE! - in poly.edu.vn.display.other.ChuyendeDaoNGTest
poly.edu.vn.display.other.ChuyendeDaoNGTest.testDelete  Time elapsed: 0.066 s  <<< FAILURE!
java.lang.AssertionError: The test case is a prototype.
	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testDelete(ChuyendeDaoNGTest.java:110)

poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithNotFoundID  Time elapsed: 0.05 s  <<< FAILURE!
org.mockito.exceptions.misusing.InvalidUseOfMatchersException: 

Misplaced or misused argument matcher detected here:

-> at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithNotFoundID(ChuyendeDaoNGTest.java:163)
-> at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithNotFoundID(ChuyendeDaoNGTest.java:164)

You cannot use argument matchers outside of verification or stubbing.
Examples of correct usage of argument matchers:
    when(mock.get(anyInt())).thenReturn(null);
    doThrow(new RuntimeException()).when(mock).someVoidMethod(anyObject());
    verify(mock).someMethod(contains("foo"))

This message may appear after an NullPointerException if the last matcher is returning an object 
like any() but the stubbed method signature expect a primitive argument, in this case,
use primitive alternatives.
    when(mock.get(any())); // bad use, will raise NPE
    when(mock.get(anyInt())); // correct usage use

Also, this error might show up because you use argument matchers with methods that cannot be mocked.
Following methods *cannot* be stubbed/verified: final/private/equals()/hashCode().
Mocking methods declared on non-public parent classes is not supported.


poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithvalidID  Time elapsed: 0.011 s  <<< FAILURE!
org.powermock.core.InvocationException: 
org.mockito.exceptions.misusing.UnfinishedStubbingException: 
Unfinished stubbing detected here:
-> at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithNotFoundID(ChuyendeDaoNGTest.java:161)

E.g. thenReturn() may be missing.
Examples of correct stubbing:
    when(mock.isOk()).thenReturn(true);
    when(mock.isOk()).thenThrow(exception);
    doThrow(exception).when(mock).someVoidMethod();
Hints:
 1. missing thenReturn()
 2. you are trying to stub a final method, which is not supported
 3: you are stubbing the behaviour of another mock inside before 'thenReturn' instruction is completed

	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithvalidID(ChuyendeDaoNGTest.java:138)
Caused by: org.mockito.exceptions.misusing.UnfinishedStubbingException: 

Unfinished stubbing detected here:
-> at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithNotFoundID(ChuyendeDaoNGTest.java:161)

E.g. thenReturn() may be missing.
Examples of correct stubbing:
    when(mock.isOk()).thenReturn(true);
    when(mock.isOk()).thenThrow(exception);
    doThrow(exception).when(mock).someVoidMethod();
Hints:
 1. missing thenReturn()
 2. you are trying to stub a final method, which is not supported
 3: you are stubbing the behaviour of another mock inside before 'thenReturn' instruction is completed

	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testFindByIdWithvalidID(ChuyendeDaoNGTest.java:138)

poly.edu.vn.display.other.ChuyendeDaoNGTest.testInsertWithEmptyModel  Time elapsed: 0.007 s  <<< FAILURE!
org.testng.TestException: 

Method ChuyendeDaoNGTest.testInsertWithEmptyModel()[pri:0, instance:poly.edu.vn.display.other.ChuyendeDaoNGTest_$$_jvste6_0@61710c6] should have thrown an exception of type class java.lang.Exception

poly.edu.vn.display.other.ChuyendeDaoNGTest.testSelect  Time elapsed: 0.008 s  <<< FAILURE!
java.lang.NullPointerException
	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testSelect(ChuyendeDaoNGTest.java:118)

poly.edu.vn.display.other.ChuyendeDaoNGTest.testTim  Time elapsed: 0.002 s  <<< FAILURE!
java.lang.AssertionError: The test case is a prototype.
	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testTim(ChuyendeDaoNGTest.java:128)

poly.edu.vn.display.other.ChuyendeDaoNGTest.testUpdate  Time elapsed: 0.003 s  <<< FAILURE!
java.lang.NullPointerException
	at poly.edu.vn.display.other.ChuyendeDaoNGTest.testUpdate(ChuyendeDaoNGTest.java:100)

